[docs] Update using-fcm.md

mrzachnugent opened the pull request:
This proposes to not use `expo` command for uploading server credentials and update missing info manually on expo.dev.

# Why

Docs recommend command `expo push:android:upload --api-key <your-token-here>` which created a new Legacy Application Identifier. This lead to android notifications not working at all. Spent a lot of time trying to understand why android notifications were not working.

<!--
Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.
-->

# How
I removed the Server Credentials from the Legacy Application Identifier which cause the android push notifications to finally work.

<!--
How did you build this feature or fix this bug and why?
-->

# Test Plan

Upload Server Credentials the old way with `expo push:android:upload --api-key <your-token-here>`. Check if android notifications stopped working.

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [ x ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ x ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ x ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

Commit:
[chore] Update CHANGELOG.md (#16232)

mrzachnugent committed:
Update using-fcm.md

Docs used command `expo push:android:upload --api-key <your-token-here>` which created a new Legacy Application Identifier. This lead to android notifications not working at all.

This proposes to remove expo command and update missing info manually on expo.dev

mrzachnugent committed:
Merge branch 'main' into patch-2

amandeepmittal commented on docs/pages/push-notifications/using-fcm.md:
```diff
@@ -65,9 +65,15 @@ In order for Expo to send notifications from our servers using your credentials,
 3. Copy the token listed next to **Server key**.
 
-> **Note:** Server Key is only available in **Cloud Messaging API (Legacy)**, which may be Disabled by default. Enable it by clicking the 3-dot menu > Manage API in Google Cloud Console and follow the flow there. Once the legacy messaging API is enabled, you should see Server Key in that section.
+  > **Note:** Server Key is only available in **Cloud Messaging API (Legacy)**, which may be Disabled by default. Enable it by clicking the 3-dot menu > Manage API in Google Cloud Console and follow the flow there. Once the legacy messaging API is enabled, you should see Server Key in that section.
```
```suggestion
> **Note:** Server Key is only available in **Cloud Messaging API (Legacy)**, which may be Disabled by default. Enable it by clicking the 3-dot menu > Manage API in Google Cloud Console and follow the flow there. Once the legacy messaging API is enabled, you should see Server Key in that section.
```

amandeepmittal commented on docs/pages/push-notifications/using-fcm.md:
```diff
@@ -65,9 +65,15 @@ In order for Expo to send notifications from our servers using your credentials,
-4. Run `expo push:android:upload --api-key <your-token-here>`, replacing `<your-token-here>` with the string you just copied. We'll store your token securely on our servers, where it will only be accessed when you send a push notification.
+4. In your [expo account](https://expo.dev/), select your project, and click on Credentials in the navigation menu. Click on your Application Identifier _(something like com.yourcompany.yourapp)_. 
+
+  > **Note:** For Legacy Application Identifiers, run `expo push:android:upload --api-key <your-token-here>`, replacing `<your-token-here>` with the string you just copied. We'll store your token securely on our servers, where it will only be accessed when you send a push notification.
```
```suggestion
> **Note:** For Legacy Application Identifiers, run `expo push:android:upload --api-key <your-token-here>`, replacing `<your-token-here>` with the string you just copied. We'll store your token securely on our servers, where it will only be accessed when you send a push notification.
```

amandeepmittal commented on docs/pages/push-notifications/using-fcm.md:
```diff
@@ -65,9 +65,15 @@ In order for Expo to send notifications from our servers using your credentials,
+  > **Note:** Server Key is only available in **Cloud Messaging API (Legacy)**, which may be Disabled by default. Enable it by clicking the 3-dot menu > Manage API in Google Cloud Console and follow the flow there. Once the legacy messaging API is enabled, you should see Server Key in that section.
 
-4. Run `expo push:android:upload --api-key <your-token-here>`, replacing `<your-token-here>` with the string you just copied. We'll store your token securely on our servers, where it will only be accessed when you send a push notification.
+4. In your [expo account](https://expo.dev/), select your project, and click on Credentials in the navigation menu. Click on your Application Identifier _(something like com.yourcompany.yourapp)_. 
```
These points are not coming up with as number 4 or number 5. I think adding the note in between might be breaking the flow of numbered list. Can you please have a look to fix this?

<img width="885" alt="CleanShot 2022-09-29 at 04 21 17@2x" src="https://user-images.githubusercontent.com/10234615/192902344-ef9f9f71-873b-4f81-8920-bc6dc9e3b315.png">

amandeepmittal replied:
```suggestion
4. In your [Expo account's dashboard](https://expo.dev/), select your project, and click on **Credentials** in the navigation menu. Then, click on your **Application Identifier** that follows the pattern:`com.company.app`.
```

amandeepmittal commented on docs/pages/push-notifications/using-fcm.md:
```diff
@@ -65,9 +65,15 @@ In order for Expo to send notifications from our servers using your credentials,
+
+5. Check that all of your Service Credentials are correctly filled in. If your FCM Server Key is empty, add the copied Server key from step 3. If the FCM Server Key is populated, verify that the last characters match the Server Key from step 3.
+
+  > **Note:** Having Service Credentials in both Legacy and non-legacy Application Identifiers can prevent push notifications from working on android devices. If you have a Legacy Application Identifier, you might need to remove all of its Service Credentials. 
```
```suggestion
> **Note:** Having Service Credentials in both Legacy and non-legacy Application Identifiers can prevent push notifications from working on android devices. If you have a Legacy Application Identifier, you might need to remove all of its Service Credentials. 
```

amandeepmittal commented on docs/pages/push-notifications/using-fcm.md:
```diff
@@ -65,9 +65,15 @@ In order for Expo to send notifications from our servers using your credentials,
+
+  > **Note:** Having Service Credentials in both Legacy and non-legacy Application Identifiers can prevent push notifications from working on android devices. If you have a Legacy Application Identifier, you might need to remove all of its Service Credentials. 
 
 That's it -- users who run this new version of the app will now receive notifications through FCM using your project's credentials. You just send the push notifications as you normally would (see [guide](sending-notifications.md)). We'll take care of choosing the correct service to send the notification.
```
```suggestion
Users who run this new version of the app will now receive notifications through FCM using your project's credentials. You can now send push notifications normally. For more information, see [Sending Notifications](/push-notifications/sending-notifications/). We'll take care of choosing the correct service to send the notification.
```

Unrelated to this PR, but I think it is nice to have this improvement.

amandeepmittal commented on docs/pages/push-notifications/using-fcm.md:
```diff
@@ -65,9 +65,15 @@ In order for Expo to send notifications from our servers using your credentials,
+
+  > **Note:** For Legacy Application Identifiers, run `expo push:android:upload --api-key <your-token-here>`, replacing `<your-token-here>` with the string you just copied. We'll store your token securely on our servers, where it will only be accessed when you send a push notification.
+
+5. Check that all of your Service Credentials are correctly filled in. If your FCM Server Key is empty, add the copied Server key from step 3. If the FCM Server Key is populated, verify that the last characters match the Server Key from step 3.
```
```suggestion
5. Check that all of your **Service Credentials** are filled correctly. If the **FCM Server Key** is empty, add the copied **Server key** from **Step 3**. If the FCM Server Key is populated, verify that the last characters match the **Server Key** from **Step 3**.
```

amandeepmittal reviewed (changes requested):
Thanks for this PR @mrzachnugent!

Left some suggestions.

mrzachnugent committed:
Merge remote-tracking branch 'origin/master' into pr/mrzachnugent/18806

mrzachnugent committed:
[docs] Update Firebase Cloud Messaging docs from reviewed suggestions

mrzachnugent committed:
[docs] Update Firebase Cloud Messaging fix typo

mrzachnugent commented:
@amandeepmittal 
Thank you for suggestion. I applied them and the numbered flow is now fixed.
**There is a conflict I cannot resolve.** 
I tried merging `origin/master` into this patch but __I did not get any conflicts locally__.
I checked out the file on master and it looks like the original file I updated.
Have you seen something like this before or is this something only you can resolve?

amandeepmittal commented on docs/pages/push-notifications/using-fcm.md:
```diff
@@ -1,6 +1,7 @@
----
+git---
```
```suggestion
---
```

amandeepmittal commented on docs/pages/push-notifications/using-fcm.md:
```diff
@@ -1,6 +1,7 @@
+git---
 title: Using FCM for Push Notifications
 sidebar_title: Using FCM
+
```
```suggestion
```

amandeepmittal commented:
@mrzachnugent, the conflict is appearing because we recently moved to MDX v2 and had to rename the file extensions from `.md` to `.mdx`. This PR was created before that 😅  (sorry for the inconvenience caused due to that).

I usually follow what GitHub suggests under "Conflicting Plugins": 
<img width="794" alt="CleanShot 2022-09-30 at 04 50 12@2x" src="https://user-images.githubusercontent.com/10234615/193158719-1dabb99f-6ceb-4653-be25-2fd1bde295d9.png">

Feel free to create a new PR if you can't get the conflicts resolved with the changes from this 😄

Simek commented on docs/pages/push-notifications/using-fcm.md:
````diff
@@ -14,37 +14,37 @@ Note that FCM is not currently available for `expo-notifications` on iOS.
-  }
-  ```
+```javascript
+{
+  ...
+  "client": [
+    {
+      "api_key": [
+        {
+          "current_key" "<your Google Cloud Platform API key>",
+        }
+      ],
+    }
+  ]
+}
+```
 
-  > **Note:** Firebase will create an API key in the Google Cloud Platform console with a name like `Android key (auto created by Firebase)`. **This is not always the same key as the one found in `google-services.json`. Always confirm your key and associated restrictions in the Google Cloud Platform console.**
+> **Note:** Firebase will create an API key in the Google Cloud Platform console with a name like `Android key (auto created by Firebase)`. **This is not always the same key as the one found in `google-services.json`. Always confirm your key and associated restrictions in the Google Cloud Platform console.**
````
Those whitespaces are intentional, and they ensure that code block and callout is nested under list entry.

amandeepmittal replied:
I didn't realize the callout was nested. Will have to check again. Thanks for pointing it out 👍

mrzachnugent replied:
@Simek Thanks for pointing that out. I forgot to save without formatting.

@amandeepmittal I made another [PR](https://github.com/expo/expo/pull/19330/files) du to problems with the change of the file extension. Thank you
